packages/tracer/src/Tracer.ts

```diff
@@ -128,7 +128,7 @@ class Tracer extends Utility implements TracerInterface {
   // envVarsService is always initialized in the constructor in setOptions()
   private envVarsService!: EnvironmentVariablesService;
   
-  private serviceName?: string;
+  private serviceName!: string;
```

Comment:
This is needed for Typescript compiler. It can't guarantee that `serviceName` is not undefined, as it only looks in the constructor and not the nested functions called in the constructor. We _know_ it will be set, but the ts compiler does not.

Reply:
Done